unittests/internaltests.py

```diff
@@ -1415,10 +1415,24 @@ def test_typed_kwarg_since_values(self) -> None:
                       since_values={list: '1.9'}),
             KwargInfo('new_dict', (ContainerTypeInfo(list, str), ContainerTypeInfo(dict, str)), default={},
                       since_values={dict: '1.1'}),
+            KwargInfo('foo', (str, ContainerTypeInfo(list, str), ContainerTypeInfo(dict, str)), default={},
+                      since_values={str: '1.1', ContainerTypeInfo(list, str): '1.2', ContainerTypeInfo(dict, str): '1.3'}),
         )
         def _(obj, node, args: T.Tuple, kwargs: T.Dict[str, str]) -> None:
             pass
 
+        with self.subTest('new string type'), mock.patch('sys.stdout', io.StringIO()) as out:
+            _(None, mock.Mock(subproject=''), [], {'foo': 'foo'})
+            self.assertRegex(out.getvalue(), r"""WARNING: Project targets '1.0'.*introduced in '1.1': "testfunc" keyword argument "foo" of type str""")
+
+        with self.subTest('new array of string type'), mock.patch('sys.stdout', io.StringIO()) as out:
+            _(None, mock.Mock(subproject=''), [], {'foo': ['foo']})
+            self.assertRegex(out.getvalue(), r'''WARNING: Project targets '1.0'.*introduced in '1.2': "testfunc" keyword argument "foo" of type "array[str]"''')
+
+        with self.subTest('new dict of string type'), mock.patch('sys.stdout', io.StringIO()) as out:
+            _(None, mock.Mock(subproject=''), [], {'foo': {'plop': 'foo'}})
+            self.assertRegex(out.getvalue(), r'''WARNING: Project targets '1.0'.*introduced in '1.3': "testfunc" keyword argument "foo" of type "dict[str]"''')
+
```

Comment:
It would be nice to have at least one deprecation test as well.

Reply:
done